pkg/azuredx/adxauth/adxcredentials/builder.go

```diff
@@ -0,0 +1,121 @@
+package adxcredentials
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/grafana/azure-data-explorer-datasource/pkg/azuredx/adxauth/adxcredentials/internal/maputil"
+	"github.com/grafana/grafana-azure-sdk-go/azcredentials"
+	"github.com/grafana/grafana-azure-sdk-go/azsettings"
+)
+
+func FromDatasourceData(data map[string]interface{}, secureData map[string]string) (azcredentials.AzureCredentials, error) {
+	var credentials azcredentials.AzureCredentials
+	var err error
+
+	credentials, err = azcredentials.FromDatasourceData(data, secureData)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fallback to legacy credentials format
+	if credentials == nil {
+		credentials, err = getFromLegacy(data, secureData)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Current implementation of on-behalf-of authentication requires OAuth token pass-thru enabled
+	switch credentials.(type) {
+	case *azcredentials.AzureClientSecretOboCredentials:
+		if err := ensureOnBehalfOfSupported(data); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
having this check separated than the actual reading of properties for OBO makes it a bit more difficult to use (or easy to forget).
Can you move this check to the `azcredentials.FromDatasourceData` and `getFromLegacy` methods? I know that is having the code twice but that way we make sure that other plugins calling `azcredentials.FromDatasourceData` also have the check.

Reply:
I was thinking about it but I'm concerned that this is implementation detail of now OBO implemented in ADX so I'm not sure. SDK currently doesn't have OBO implementation, only definition of the credentials.

Can we revisit it when/if we decide to move OBO to SDK?

Reply:
we are in a weird state in which _parts_ of the OBO implementation are in the SDK and some others don't. I would at least try to complete the parts that are already in the SDK.

In this case, properties parsing happen in the SDK and since the SDK are reading the OBO settings, I still think that this check belongs to the SDK.

Reply:
SDK only has declaration for `azcredentials.AzureClientSecretOboCredentials` and a method to parse it, that's it.

The fact that ADX implementation needs `oauthPassThru` is implementation detail of ADX. SDK has nothing to do with headers.